### PR TITLE
Fix #4235: elastic_transform2d identity mapping with align_corners=False

### LINKED ISSUE #4235: [Bug]: elastic_transform2d is not identity for zero displacement with align_corners=False
## What happened?

`elastic_transform2d` does not preserve the input image when the displacement field is zero and `align_corners=False` is used.

A zero displacement field represents an identity deformation, so the output should be equal to the input up to floating-point tolerance.

The current implementation constructs the sampling grid using `create_meshgrid(h, w)` and then passes it to `torch.nn.functional.grid_sample` with the selected `align_corners` value.

Since `create_meshgrid` produces the corner-aligned normalized `[-1, 1]` grid, this grid is not consistent with the coordinate convention used by `grid_sample` when `align_corners=False`.

As a result, even with an exactly zero displacement field, the image is resampled at shifted coordinates.

### Expected behavior

With:

python
noise = torch.zeros(B, 2, H, W)

the transformation should be an identity transformation:
elastic_transform2d(image, noise, ...) == image
within numerical tolerance.

Actual behavior

On the current main checkout, using a 4x4 deterministic input and a zero displacement field produces a maximum absolute difference of:
tensor(11.2500)
So the input is substantially changed even though the displacement is exactly zero.

This behavior is specific to the align_corners=False coordinate convention; the same corner-aligned grid is consistent with align_corners=True.

### Minimal reproduction

```python
import torch
from kornia.geometry.transform import elastic_transform2d

image = torch.arange(16.0).reshape(1, 1, 4, 4)
noise = torch.zeros(1, 2, 4, 4)

output = elastic_transform2d(
    image,
    noise,
    kernel_size=(3, 3),
    sigma=(1.0, 1.0),
    alpha=(1.0, 1.0),
)

print("Input:")
print(image)

print("\nOutput:")
print(output)

print("\nMax absolute difference:")
print((output - image).abs().max())
```

### Environment

Kornia 0.9.0rc1 | PyTorch 2.14.0+cpu | Python 3.13.11 | Windows

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dhruvv16-hash**
Fixes #4235

### Description

This PR fixes a bug in `elastic_transform2d` where a zero displacement field does not result in an identity transformation when `align_corners=False`.

The issue was caused by `elastic_transform2d` always using `create_meshgrid(normalized_coordinates=True)` to generate the base grid, which assumes the `align_corners=True` convention (`[-1, 1]` mapped to outer pixel centers). When this grid was passed into `F.grid_sample` with `align_corners=False`, the coordinate meaning shifted, resulting in the image being slightly resampled at shifted coordinates.

I fixed this by explicitly generating the grid according to the selected `align_corners` flag so that it's consistent with `grid_sample` expectations. 

Includes tests that verify identity mappings for both `align_corners=False` and `align_corners=True`.

### CODE REVIEW

**Review by ducha-aiki — CHANGES_REQUESTED** [review]
Reviewed exact head `eb1782b96a95dc2848ad04350d0a583540f5a2b0`.

The identity-grid correction looks sound at ordinary precision, and the complete elastic-transform file passed 36 CPU float32/float64 tests. This targets the same defect as #4382, which has already addressed its review requests; I recommend consolidating on that PR rather than merging both implementations.

[P2] If this PR is retained, add the required Unreleased Bug fixes entry citing #4403. This changes public elastic-transform behavior, but the current diff has no CHANGELOG entry. Hosted float16 CI is also cancelled and collector is red, so successful replacement checks are needed.

Nonblocking precision limit: both this PR and #4382 still show zero-displacement differences for sufficiently wide float16/bfloat16 images because normalized sampling coordinates are quantized. Comparison with current main and #4382 did not establish a new regression specific to this implementation; that limitation should be handled separately.

Validation used isolated checkouts with import origins verified. No fresh whole-suite, CUDA or MPS run was performed locally.

Posted on behalf of [@ducha-aiki](https://github.com/ducha-aiki) by Codex (gpt-6-astra).

**Comment by dhruvv16-hash** [comment]
Closing as this was addressed and merged in #4382. Thank you!
